EDU-129 Clarifies why you use Memos (#2060)
temporalio/documentation: docs-src/concepts/what-is-a-memo.md

id: what-is-a-memo
title: What is a Memo?
sidebar_label: Memo
description: A Memo is a non-indexed user-supplied set of Workflow Execution metadata that is displayed with filtered list results.
tags:
  - term
  - explanation
---

> A Memo is a non-indexed set of Workflow Execution metadata that users supply and that a filtered list displays.
> Use Memos to define metadata supplied by users in a filtered list.
> 
> The primary purpose of using a Memo is to enhance the organization and management of Workflow Executions.
> Add your own metadata, such as notes or descriptions, to a Workflow Execution, which lets you annotate and categorize Workflow Executions based on user-defined criteria.

Reviewer: ```suggestion
A Memo is a non-indexed set of Workflow Execution metadata that you supply at start time or in Workflow code and that is returned when you describe or list Workflow Executions.

The primary purpose of using a Memo is to enhance the organization and management of Workflow Executions.
Add your own metadata, such as notes or descriptions, to a Workflow Execution, which lets you annotate and categorize Workflow Executions based on criteria you define.
```

- clarify user = developer / the reader
- the two API calls that return memos are describe and list

Reviewer: execute/start is the "at start time". the returned values, in describe and list, might be --fields in the CLI commands. it's also in the gRPC API and SDK APIs.

This feature is particularly useful when dealing with numerous Workflow Executions because it facilitates the addition of context, reminders, or any other relevant information that aids in understanding or tracking the Workflow Execution.

<!--

:::note Use Memos judiciously

Memos shouldn't store data that's critical to the execution of a Workflow, for some of the following reasons:

- Unlike Workflow inputs, Memos lack type safety
- Memos are subject to eventual consistency and may not be immediately available

> - Excessive reliance on memos hides mutable state from the Workflow Execution History

Reviewer: ```suggestion
- Excessive reliance on Memos hides mutable state from the Workflow Execution History
```

  - explanation
---

> A Memo is a non-indexed set of Workflow Execution metadata that users supply and that the Filtered List displays. Use Memos to define metadata supplied by users in a Filtered List.

Reviewer 2: ```suggestion
A Memo is a non-indexed set of Workflow Execution metadata that users supply and that a Filtered List displays.
Use Memos to define metadata supplied by users in a Filtered List.
```

> The primary purpose of using a memo is to enhance the organization and management of Workflow Executions. Add your own metadata, such as notes or descriptions, to a Workflow Execution, which enables you to annotate and categorize Workflow Executions based on user-defined criteria. This feature proves particularly useful when dealing with numerous Workflow Executions as it facilitates the addition of context, reminders, or any other relevant information that aids in understanding or tracking the Workflow Execution.

Reviewer 2: ```suggestion
The primary purpose of using a memo is to enhance the organization and management of Workflow Executions.
Add your own metadata, such as notes or descriptions, to a Workflow Execution, which lets you annotate and categorize Workflow Executions based on user-defined criteria.
This feature is particularly useful when dealing with numerous Workflow Executions because it facilitates the addition of context, reminders, or any other relevant information that aids in understanding or tracking the Workflow Execution.
```
- For "enables → lets", see https://developers.google.com/style/word-list#enable.
- For "as → because", see https://developers.google.com/style/word-list#as.

  - explanation
---

> A Memo is a non-indexed set of Workflow Execution metadata that users supply and that a Filtered List displays.

Reviewer: what's a filtered list? a list of WEs that comes back from ListWorkflows with a List Filter? i'd link to definition

"user" sounds to me like end user, is that what you mean? do we use the term elsewhere in the docs? it could also be programmatically supplied by the dev

> Use Memos to define metadata supplied by users in a Filtered List.

Reviewer: i'm not getting anything more than i did in the sentence prior, perhaps remove?

> The primary purpose of using a memo is to enhance the organization and management of Workflow Executions.

Reviewer 3: ```suggestion
The primary purpose of using a Memo is to enhance the organization and management of Workflow Executions.
```

> Add your own metadata, such as notes or descriptions, to a Workflow Execution, which lets you annotate and categorize Workflow Executions based on user-defined criteria.

Reviewer: same note—not sure what a user is
